components/admin_console/permission_schemes_settings/permission_group.tsx

```diff
@@ -114,16 +122,16 @@ export default class PermissionGroup extends React.PureComponent {
         } else {
             const permissionsToToggle = [];
             for (const permission of getRecursivePermissions(permissions)) {
-                if (role.permissions.indexOf(permission) === -1 && !this.fromParent(permission)) {
+                if (role?.permissions?.indexOf(permission) === -1 && !this.fromParent(permission)) {
                     permissionsToToggle.push(permission);
                 }
             }
-            this.setState({prevPermissions: role.permissions, expanded: false});
+            this.setState({prevPermissions: role && role.permissions ? role.permissions : [], expanded: false});
```

Comment:
Conventionally we would use an early return in the IF statement if the role is not set, you can also assign the empty array as default value in the state.